kapitan/refs/secrets/vaultkv.py

```diff
@@ -0,0 +1,285 @@
+    """
+
+    def __init__(self, data, **kwargs):
+        """
+        Set vault parameter and encoding of data
+        """
+        self.data = data.rstrip()
+        self.parameter = kwargs.get('parameter')
+        super().__init__(self.data,**kwargs)
+        self.type_name = 'vaultkv'
+
+    @classmethod
+    def from_params(cls, data, ref_params):
+        """
+        Return new VaultSecret from data and ref_params: target_name
+        parameters will be grabbed from the inventory via target_name
+        """
+        try:
+            target_name = ref_params.kwargs['target_name']
+            if target_name is None:
+                raise ValueError('target_name not set')
+
+            target_inv = cached.inv['nodes'].get(target_name, None)
+            if target_inv is None:
+                raise ValueError('target_inv not set')
+
+            ref_params.kwargs['parameter'] = target_inv['parameters']['kapitan']['secrets']['vaultkv']
+            return cls(data, **ref_params.kwargs)
+        except KeyError:
+            raise RefError("Could not create VaultSecret: vaultkv parameters missing")
+
+    @classmethod
+    def from_path(cls, ref_full_path, **kwargs):
+        """
+        return a new Ref from file at ref_full_path
+        the data key in the file must be base64 encoded
+        """
+        try:
+            with open(ref_full_path) as fp:
+                obj = yaml.load(fp, Loader=YamlLoader)
+                _kwargs = {key: value for key, value in obj.items() if key not in ('data', 'from_base64')}
+                inv = inventory_reclass(kwargs['inventory'])
+                target_inv = inv['nodes'][kwargs['target_name']]
+                if target_inv is None:
+                    raise ValueError('target not set')
+
+                _kwargs['parameter'] = target_inv['parameters']['kapitan']['secrets']['vaultkv']
+                kwargs.update(_kwargs)
+                return cls(obj['data'], from_base64=True, **kwargs)
+
+        except IOError as ex:
+            if ex.errno == errno.ENOENT:
+                return None
+
+        except KeyError:
+            raise VaultError("Could not fetch VaultSecret: vaultkv parameters missing")
+            return None
+
+    def reveal(self):
+        """
+        Returns decrypted data
+        """
+        # can't use super().reveal() as we want bytes
+        try:
+            if self.encoding == 'base64':
+                self.data = base64.b64decode(self.data, validate=True)
+
+            ref_data = base64.b64decode(self.data, validate=True)
+        except b_error:
+            exit(
+                "non-alphabet characters in the data"
+            )
+
+        return self._decrypt(ref_data)
+
+    def _decrypt(self, data):
+        """
+        Authenticate with Vault server & returns value of the key from secret
+
+        :returns: secret in plain text
+        """
+        try:
+            client = vault_obj(self.parameter)
```

Comment:
Could you make a small new class in this file with the VaultParameters instead of passing an object with an expected structure stated in comments? That way you can do all the evaluations of whether specific tokens exist in that constructor. :)